src/common/com/intellij/plugins/haxe/model/HaxeImportModel.java

```diff
@@ -0,0 +1,154 @@
+    return alias.getText();
+  }
+
+  public HaxeReferenceExpression getReferenceExpression() {
+    return basePsi.getReferenceExpression();
+  }
+
+  @NotNull
+  @Override
+  public List<HaxeModel> getExposedMembers() {
+    FullyQualifiedInfo qualifiedInfo = getQualifiedInfo();
+    List<HaxeModel> result = null;
+    if (hasWildcard()) {
+      if (qualifiedInfo.memberName != null) return Collections.emptyList();
+
+      if (qualifiedInfo.fileName != null && qualifiedInfo.className == null) {
+        qualifiedInfo = new FullyQualifiedInfo(qualifiedInfo.packagePath, qualifiedInfo.fileName, qualifiedInfo.fileName, null);
+      }
+      List<HaxeModel> items = HaxeProjectModel.fromElement(basePsi).resolve(qualifiedInfo, basePsi.getResolveScope());
+      if (items != null && items.size() > 0) {
+        result = items.stream()
+          .filter(model -> model instanceof HaxeExposableModel)
+          .flatMap(model -> ((HaxeExposableModel)model).getExposedMembers().stream())
+          .collect(Collectors.toList());
+      }
+    }
+    else {
+      if (hasAlias() && qualifiedInfo.fileName != null && qualifiedInfo.className == null) {
+        qualifiedInfo = new FullyQualifiedInfo(qualifiedInfo.packagePath, qualifiedInfo.fileName, qualifiedInfo.fileName, null);
+      }
+      result = HaxeProjectModel.fromElement(basePsi).resolve(qualifiedInfo, basePsi.getResolveScope());
+      if (result != null && !result.isEmpty() && (result.get(0) instanceof HaxeFileModel || result.get(0) instanceof HaxePackageModel)) {
+        result = ((HaxeExposableModel)result.get(0)).getExposedMembers();
+      }
+    }
+
+    return result != null ? result : Collections.emptyList();
+  }
+
+  @Nullable
+  public PsiElement exposeByName(String name) {
+    if (name == null) return null;
+
+    if (hasWildcard()) {
+      for (HaxeModel exposedMember : getExposedMembers()) {
+        if (Objects.equals(exposedMember.getName(), name)) return exposedMember.getBasePsi();
+      }
+    }
+    else {
+      if (getReferenceExpression() != null) {
+        FullyQualifiedInfo qualifiedInfo = getQualifiedInfo();
+
+        if (equalsToAlias(name) || qualifiedInfo.equalsToNamedPart(name) || getExposedMember(name) != null) {
+          HaxeModel result = getExposedMember(name);
+          return result != null ? result.getBasePsi() : null;
+        }
+      }
+    }
+
+    return null;
+  }
+
+  private boolean equalsToAlias(String name) {
+    return hasAlias() && Objects.equals(getAliasName(), name);
+  }
+
+  private HaxeModel getExposedMember(String name) {
+    List<HaxeModel> members = getExposedMembers();
+    if (members.isEmpty()) return null;
+    if (hasAlias()) {
+      return (Objects.equals(getAliasName(), name)) ? members.get(0) : null;
+    }
+    return members.stream()
+      .filter(model -> model.getName().equals(name))
+      .findFirst()
+      .orElse(null);
+  }
+
+  @Override
+  public String getName() {
+    return null;
```

Comment:
Why should this be null instead of the name from the FQI?

Reply:
Because imports don't have their own fully qualified info? You can't address to `some.packagename.of.the.import` and get any import statement by it. That's the meaning of fully qualified info, purpose of it's existing can't be switched for situation.

Reply:
OK.  So this confuses me a bit.  It's probably my thinking about how the HaxeImportModel is intended to be used.  I expect it to be used something like this:

```Haxe
HaxeFileModel file = new HaxeFileModel(psiElement.getVirtualFile().getName());
List<HaxeImportModel> imports = file.getImports();
for (HaxeImportModel import : imports) {
    Sys.out.println(import.getName());
    [... other, more interesting stuff]
}
```